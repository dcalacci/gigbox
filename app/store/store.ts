import { createStore } from 'redux';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
// import logger from 'redux-logger'
// import thunk from 'redux-thunk'
import {
    persistStore,
    persistReducer,
    FLUSH,
    REHYDRATE,
    PAUSE,
    PERSIST,
    PURGE,
    REGISTER,
} from 'redux-persist';
import AsyncStorage from '@react-native-async-storage/async-storage';

import rootReducer from '.';

const persistConfig = {
    key: 'gigbox',
    storage: AsyncStorage,
    blacklist: ['otp'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
    reducer: persistedReducer,
    middleware: getDefaultMiddleware({
        // ignore persist actions, which serialize functions. @reduxjs/toolkit throws a warning when
        // you serialize non-basic types
        serializableCheck: {
            ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
        },
    }),
});

const persistor = persistStore(store);

export { store, persistor };
