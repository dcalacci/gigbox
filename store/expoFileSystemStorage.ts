import * as FileSystem from 'expo-file-system';
import type { Storage } from 'redux-persist';

// interface Storage {
//   getItem(key: string): Promise<string>;
//   setItem(key: string, value: string): Promise<string | void>;
//   removeItem(key: string): Promise<void>;
// }

function generateFolderPath() {
    return FileSystem.documentDirectory + 'appData/';
}

function generateFilePath(key: string) {
    const fileName = key.replace(/[^a-z0-9.\-_]/gi, '-');
    return generateFolderPath() + fileName;
}

function writeFile(path: string, value: string) {
    FileSystem.writeAsStringAsync(path, value);
}

export class ExpoFileSystemStorage implements Storage {
    getItem(key: string) {
        return FileSystem.readAsStringAsync(generateFilePath(key));
    }

    setItem(key: string, value: string): Promise<string | void> {
        const folderPath = generateFolderPath();
        return FileSystem.getInfoAsync(folderPath).then((info) => {
            const filePath = generateFilePath(key);
            if (info.exists) {
                writeFile(filePath, value);
            } else {
                FileSystem.makeDirectoryAsync(folderPath, {
                    intermediates: true,
                }).then(() => writeFile(filePath, value));
            }
        });
    }

    removeItem(key: string) {
        return FileSystem.deleteAsync(generateFilePath(key), { idempotent: true });
    }
}
