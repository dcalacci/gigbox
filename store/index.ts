import { combineReducers } from "redux";
import ClockSlice from '../features/clock/clockSlice'
import AuthSlice from '../features/auth/authSlice'
import OtpSlice from '../features/auth/otpSlice'

const rootReducer = combineReducers({
  clock: ClockSlice,
  auth: AuthSlice,
  otp: OtpSlice
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
