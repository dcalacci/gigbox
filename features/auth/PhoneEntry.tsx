import React, { useState, useEffect, useRef } from 'react';
import { TextInput, Text, Pressable } from 'react-native';
import { useSelector, useDispatch } from 'react-redux';
import { SafeAreaView } from 'react-native-safe-area-context';
import { tailwind } from 'tailwind';
import Toast from 'react-native-root-toast';
import { AsYouType, parsePhoneNumber } from 'libphonenumber-js';
import { RootState } from '../../store/index';
import { requestOtp, loginWithOtp, reset, clearErrorMessage } from './otpSlice';
import { useQueryClient } from 'react-query';

const PhoneEntry: React.FC = (props) => {
    const [phone, setPhone] = useState<string>('');
    const [phoneIsValid, setPhoneValid] = useState<boolean>(false);
    const [otp, setOtp] = useState<string>('');
    const tokenSent = useSelector((state: RootState): boolean => state.otp.tokenSent);
    const errormsg = useSelector((state: RootState): string => state.otp.errorMessage);
    const otpIsLoading = useSelector((state: RootState): boolean => state.otp.isLoading);
    const queryClient = useQueryClient();
    const dispatch = useDispatch();
    const submittedPhone = useSelector((state: RootState): string => state.otp.phone);

    const ayt = new AsYouType('US');

    useEffect(() => {
        if (errormsg != '')
            //TODO: figure out how to type this
            Toast.show(errormsg);
        dispatch(clearErrorMessage());
    }, [errormsg]);

    // validate phone number
    useEffect(() => {
        try {
            const phoneNumber = parsePhoneNumber(phone, 'US');
            setPhoneValid(phoneNumber.isValid());
        } catch (e) {
            setPhoneValid(false);
        }
    }, [phone]);

    const setFormattedPhone = (phone: string) => {
        // formats phone number to US-centric (xxx) xxx-xxxx
        console.log('setting phone:', phone);
        // only do for over length 4 -- otherwise, when you try to delete (XXX, it formats
        // incorrectly.
        if (phone.length > 4) {
            setPhone(ayt.input(phone));
        } else {
            setPhone(phone);
        }
    };

    console.log('otp is loading? ', otpIsLoading);
    return (
        <SafeAreaView style={tailwind('px-10')}>
            {tokenSent || otpIsLoading ? (
                <SafeAreaView>
                    <Text style={tailwind('text-black text-xl font-semibold')}>
                        Verify your Passcode
                    </Text>
                    <Text style={tailwind('text-gray-400 text-sm font-semibold pb-2')}>
                        Enter the passcode we sent to your phone below:
                    </Text>
                    <TextInput
                        style={[
                            tailwind('items-center py-2 px-2 rounded-md border-4'),
                            tokenSent ? tailwind('border-green-500') : null,
                        ]}
                        value={otp}
                        onChangeText={setOtp}
                        placeholder={'555555'}
                        keyboardType="numeric"
                        autoFocus={true}
                    ></TextInput>
                    <Pressable
                        style={tailwind('items-center rounded-md py-2 w-full mt-6 bg-green-500')}
                        onPress={() => {
                            queryClient.refetchQueries('loggedIn');
                            dispatch(loginWithOtp({ phone: submittedPhone, otp }));
                        }}
                    >
                        <Text style={tailwind('text-white font-semibold')}>Verify</Text>
                    </Pressable>
                    <Pressable
                        style={tailwind('items-center rounded-md py-2 w-full mt-6 bg-gray-800')}
                        onPress={() => {
                            Toast.show(`Sending code to ${submittedPhone}`);
                            dispatch(requestOtp(submittedPhone));
                        }}
                    >
                        <Text style={tailwind('text-white font-semibold')}>Re-Send Passcode</Text>
                    </Pressable>
                    <Pressable
                        style={tailwind('rounded-md mt-5')}
                        onPress={() => dispatch(reset())}
                    >
                        <Text style={tailwind('text-gray-400 text-sm font-semibold p-2')}>
                            Entered the wrong number? Tap here to restart.
                        </Text>
                    </Pressable>
                </SafeAreaView>
            ) : (
                <SafeAreaView>
                    <Text style={tailwind('text-green-500 text-3xl font-bold')}>
                        Login to Gigbox
                    </Text>
                    <Text style={tailwind('text-green-500 text-lg font-bold')}>
                        Your phone number is never saved and is only used to log you in
                    </Text>

                    <Text style={tailwind('text-black text-xl font-semibold pt-20')}>
                        Get your One-Time Passcode
                    </Text>
                    <Text style={tailwind('text-gray-400 text-sm font-semibold pb-2')}>
                        Enter your phone number below:
                    </Text>
                    <TextInput
                        testID="phone-input"
                        style={[
                            tailwind('items-center py-2 px-2 rounded-md border-4'),
                            tokenSent ? tailwind('border-green-500') : null,
                        ]}
                        key={'phoneInput'}
                        value={phone}
                        onChangeText={setFormattedPhone}
                        placeholder={'(555) 555-5555'}
                        keyboardType={'phone-pad'}
                        autoCompleteType={'tel'}
                        autoFocus={true}
                    ></TextInput>
                    <Pressable
                        testID="request-code-button"
                        style={[
                            tailwind('items-center rounded-md py-2 w-full mt-6'),
                            phoneIsValid ? tailwind('bg-green-500') : tailwind('bg-gray-600'),
                        ]}
                        disabled={!phoneIsValid}
                        onPress={() => {
                            Toast.show(`Sending code to ${phone}`);
                            dispatch(requestOtp(phone));
                        }}
                    >
                        <Text
                            testID="request-code-button-text"
                            style={tailwind('text-white font-semibold')}
                        >
                            {!phoneIsValid ? 'Enter a Valid Number' : 'Request Code'}
                        </Text>
                    </Pressable>
                </SafeAreaView>
            )}
        </SafeAreaView>
    );
};

export default PhoneEntry;
