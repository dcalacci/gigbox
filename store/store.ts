import { createStore } from "redux";
import { persistStore, persistReducer } from "redux-persist";
import { ExpoFileSystemStorage } from "./expoFileSystemStorage";

import rootReducer from ".";
const fsStorage = new ExpoFileSystemStorage();

const persistConfig = {
  key: "gigbox",
  storage: fsStorage,
};

// const reduxDevTools = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const persistedReducer = persistReducer(persistConfig, rootReducer);
const store = createStore(persistedReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());

const persistor = persistStore(store);

export { store, persistor };
