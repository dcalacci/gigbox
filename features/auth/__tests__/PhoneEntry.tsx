import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';
import { render, fireEvent, cleanup } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import { QueryClientProvider, QueryClient } from 'react-query';
const queryClient = new QueryClient();
const qMount = (c) => <QueryClientProvider client={queryClient}>{c}</QueryClientProvider>;

afterEach(cleanup);

import PhoneEntry from '../PhoneEntry';

describe('<PhoneEntry/>', () => {
    it('formats phone number corectly when entered', () => {
        const { getByTestId, toJSON} = render(
            qMount(
                <Provider store={store}>
                    <PhoneEntry />
                </Provider>
            )
        );
        expect(toJSON()).toMatchSnapshot()
        const phoneInput = getByTestId('phone-input')
        /* expect(getByTestId('phone-input').textContent).tobe(''); */
        fireEvent.changeText(phoneInput, "5555555555")
        expect(phoneInput.props.value).toBe("(555) 555-5555")
    });

    it('disables submit button if phone number is not correctly formatted', () => {
        const { getByTestId, toJSON} = render(
            qMount(
                <Provider store={store}>
                    <PhoneEntry />
                </Provider>
            )
        );
        expect(toJSON()).toMatchSnapshot()
        const phoneInput = getByTestId('phone-input')
        /* expect(getByTestId('phone-input').textContent).tobe(''); */
        fireEvent.changeText(phoneInput, "55564333333322")
        expect(phoneInput.props.value).toBe("55564333333322")
        const submitText = getByTestId("request-code-button-text")
        expect(submitText.props.children).toBe("Enter a Valid Number")
    });

    it('enables submit button if phone number is correctly formatted', () => {
        const { getByTestId, toJSON} = render(
            qMount(
                <Provider store={store}>
                    <PhoneEntry />
                </Provider>
            )
        );
        expect(toJSON()).toMatchSnapshot()
        const phoneInput = getByTestId('phone-input')
        /* expect(getByTestId('phone-input').textContent).tobe(''); */
        fireEvent.changeText(phoneInput, "7325630288")
        const submitText = getByTestId("request-code-button-text")
        expect(submitText.props.children).toBe("Request Code")
    });



    /* it('formats a phone number when entered', () => { */
    /*     const wrapper = mount( */
    /*         qMount( */
    /*             <Provider store={store}> */
    /*                 <PhoneEntry /> */
    /*             </Provider> */
    /*         ) */
    /*     ); */

    /*     expect(wrapper).toMatchSnapshot(); */

    /*     wrapper.find('TextInput').simulate('change', { target: { value: '5555555555' } }); */
    /*     expect(wrapper.find('TextInput').instance().value == '(555) 555-5555'); */

    /*     /1* act(() => { *1/ */
    /*     /1*     const wrapper = component.root; *1/ */
    /*     /1*     wrapper.findByType(TextInput).props.onChangeText('5555555555'); *1/ */
    /*     /1* }); *1/ */

    /*     /1* tree.props.setFormattedPhone("5555555555") *1/ */

    /*     expect(wrapper).toMatchSnapshot(); */
    /* }); */
});
