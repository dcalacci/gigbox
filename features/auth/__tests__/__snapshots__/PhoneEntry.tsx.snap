// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneEntry/> disables submit butten if phone number is not correctly formatted 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "paddingLeft": 40,
      "paddingRight": 40,
    }
  }
>
  <RNCSafeAreaView>
    <Text
      style={
        Object {
          "color": "rgba(16, 185, 129, 1)",
          "fontSize": 30,
          "fontWeight": "700",
          "lineHeight": 36,
        }
      }
    >
      Login to Gigbox
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(16, 185, 129, 1)",
          "fontSize": 18,
          "fontWeight": "700",
          "lineHeight": 28,
        }
      }
    >
      Your phone number is never saved and is only used to log you in
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(0, 0, 0, 1)",
          "fontSize": 20,
          "fontWeight": "600",
          "lineHeight": 28,
          "paddingTop": 80,
        }
      }
    >
      Get your One-Time Passcode
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(156, 163, 175, 1)",
          "fontSize": 14,
          "fontWeight": "600",
          "lineHeight": 20,
          "paddingBottom": 8,
        }
      }
    >
      Enter your phone number below:
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCompleteType="tel"
      autoFocus={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="(555) 555-5555"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderBottomWidth": 4,
            "borderLeftWidth": 4,
            "borderRightWidth": 4,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "borderTopWidth": 4,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
          null,
        ]
      }
      testID="phone-input"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "marginTop": 24,
            "paddingBottom": 8,
            "paddingTop": 8,
            "width": "100%",
          },
          Object {
            "backgroundColor": "rgba(75, 85, 99, 1)",
          },
        ]
      }
      testID="request-code-button"
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontWeight": "600",
          }
        }
        testID="request-code-button-text"
      >
        Enter a Valid Number
      </Text>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaView>
`;

exports[`<PhoneEntry/> disables submit button if phone number is not correctly formatted 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "paddingLeft": 40,
      "paddingRight": 40,
    }
  }
>
  <RNCSafeAreaView>
    <Text
      style={
        Object {
          "color": "rgba(16, 185, 129, 1)",
          "fontSize": 30,
          "fontWeight": "700",
          "lineHeight": 36,
        }
      }
    >
      Login to Gigbox
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(16, 185, 129, 1)",
          "fontSize": 18,
          "fontWeight": "700",
          "lineHeight": 28,
        }
      }
    >
      Your phone number is never saved and is only used to log you in
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(0, 0, 0, 1)",
          "fontSize": 20,
          "fontWeight": "600",
          "lineHeight": 28,
          "paddingTop": 80,
        }
      }
    >
      Get your One-Time Passcode
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(156, 163, 175, 1)",
          "fontSize": 14,
          "fontWeight": "600",
          "lineHeight": 20,
          "paddingBottom": 8,
        }
      }
    >
      Enter your phone number below:
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCompleteType="tel"
      autoFocus={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="(555) 555-5555"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderBottomWidth": 4,
            "borderLeftWidth": 4,
            "borderRightWidth": 4,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "borderTopWidth": 4,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
          null,
        ]
      }
      testID="phone-input"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "marginTop": 24,
            "paddingBottom": 8,
            "paddingTop": 8,
            "width": "100%",
          },
          Object {
            "backgroundColor": "rgba(75, 85, 99, 1)",
          },
        ]
      }
      testID="request-code-button"
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontWeight": "600",
          }
        }
        testID="request-code-button-text"
      >
        Enter a Valid Number
      </Text>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaView>
`;

exports[`<PhoneEntry/> enables submit button if phone number is correctly formatted 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "paddingLeft": 40,
      "paddingRight": 40,
    }
  }
>
  <RNCSafeAreaView>
    <Text
      style={
        Object {
          "color": "rgba(16, 185, 129, 1)",
          "fontSize": 30,
          "fontWeight": "700",
          "lineHeight": 36,
        }
      }
    >
      Login to Gigbox
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(16, 185, 129, 1)",
          "fontSize": 18,
          "fontWeight": "700",
          "lineHeight": 28,
        }
      }
    >
      Your phone number is never saved and is only used to log you in
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(0, 0, 0, 1)",
          "fontSize": 20,
          "fontWeight": "600",
          "lineHeight": 28,
          "paddingTop": 80,
        }
      }
    >
      Get your One-Time Passcode
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(156, 163, 175, 1)",
          "fontSize": 14,
          "fontWeight": "600",
          "lineHeight": 20,
          "paddingBottom": 8,
        }
      }
    >
      Enter your phone number below:
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCompleteType="tel"
      autoFocus={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="(555) 555-5555"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderBottomWidth": 4,
            "borderLeftWidth": 4,
            "borderRightWidth": 4,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "borderTopWidth": 4,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
          null,
        ]
      }
      testID="phone-input"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "marginTop": 24,
            "paddingBottom": 8,
            "paddingTop": 8,
            "width": "100%",
          },
          Object {
            "backgroundColor": "rgba(75, 85, 99, 1)",
          },
        ]
      }
      testID="request-code-button"
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontWeight": "600",
          }
        }
        testID="request-code-button-text"
      >
        Enter a Valid Number
      </Text>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaView>
`;

exports[`<PhoneEntry/> formats phone number corectly when entered 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "paddingLeft": 40,
      "paddingRight": 40,
    }
  }
>
  <RNCSafeAreaView>
    <Text
      style={
        Object {
          "color": "rgba(16, 185, 129, 1)",
          "fontSize": 30,
          "fontWeight": "700",
          "lineHeight": 36,
        }
      }
    >
      Login to Gigbox
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(16, 185, 129, 1)",
          "fontSize": 18,
          "fontWeight": "700",
          "lineHeight": 28,
        }
      }
    >
      Your phone number is never saved and is only used to log you in
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(0, 0, 0, 1)",
          "fontSize": 20,
          "fontWeight": "600",
          "lineHeight": 28,
          "paddingTop": 80,
        }
      }
    >
      Get your One-Time Passcode
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(156, 163, 175, 1)",
          "fontSize": 14,
          "fontWeight": "600",
          "lineHeight": 20,
          "paddingBottom": 8,
        }
      }
    >
      Enter your phone number below:
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCompleteType="tel"
      autoFocus={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="(555) 555-5555"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderBottomWidth": 4,
            "borderLeftWidth": 4,
            "borderRightWidth": 4,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "borderTopWidth": 4,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
          null,
        ]
      }
      testID="phone-input"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "marginTop": 24,
            "paddingBottom": 8,
            "paddingTop": 8,
            "width": "100%",
          },
          Object {
            "backgroundColor": "rgba(75, 85, 99, 1)",
          },
        ]
      }
      testID="request-code-button"
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontWeight": "600",
          }
        }
        testID="request-code-button-text"
      >
        Enter a Valid Number
      </Text>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaView>
`;
